tokenizers/src/models/wordlevel/mod.rs

```diff
@@ -177,7 +177,10 @@ impl Model for WordLevel {
                 .get(&*token)
                 .or_else(|| self.vocab.get(&*self.unk_token))
                 .ok_or(Error::MissingUnkToken)?,
-            value: token.to_owned(),
+            value: self
+                .vocab
+                .get(&*token)
+                .map_or(self.unk_token.clone(), |_| token.to_owned()),
```

Comment:
Other option which feel slightly cleaner IMO:

```rust
        if let Some(&id) = self.vocab.get(token) {
            Ok(vec![Token {
                id,
                value: token.to_owned(),
                offsets: (0, token.len()),
            }])
        } else if let Some(&unk_id) = self.vocab.get(&self.unk_token) {
            Ok(vec![Token {
                id: unk_id,
                value: self.unk_token.to_owned(),
                offsets: (0, token.len()),
            }])
        } else {
            Err(Box::new(Error::MissingUnkToken))
        }
```
It's personal taste, probably gets compiled the exact same way.

Reply:
indeed, it looks cleaner
in commit https://github.com/huggingface/tokenizers/pull/870/commits/a566cde962a47e331eeb7d8cff472c03784e1432,
1. made the suggested change
2. added a unit cases that tests the error

Reply:
Perfect, I'll go ahead an merge.